spread.yaml

```diff
@@ -51,6 +51,7 @@ backends:
         key: "$(HOST: echo $SPREAD_GOOGLE_KEY)"
         location: computeengine/us-east1-b
         halt-timeout: 2h
+        storage: preserve-size
```

Comment:
I am concerned about defining it at backend level, some imported images could be small and if we don't resize them to 10GB tests could fail. Currently all our images are 10GB because we are resizing them when we install all the snapd tests dependencies.

Reply:
If I do this:
```diff --git a/spread.yaml b/spread.yaml
index b2790aca7..7dfe5289d 100644
--- a/spread.yaml
+++ b/spread.yaml
@@ -51,7 +51,6 @@ backends:
         key: "$(HOST: echo $SPREAD_GOOGLE_KEY)"
         location: computeengine/us-east1-b
         halt-timeout: 2h
-        storage: preserve-size
         systems:
             - ubuntu-14.04-64:
                 workers: 6
@@ -92,6 +91,7 @@ backends:

             - amazon-linux-2-64:
                 workers: 4
+                storage: preserve-size

     google-sru:
         type: google
```
The setting is not applied:
```2018-07-27 17:42:55 Found /home/maciek/work/canonical/workspace/src/github.com/snapcore/snapd/spread.yaml.                                       
2018-07-27 17:43:01 Project content is packed for delivery (3.00MB).
2018-07-27 17:43:01 If killed, discard servers with: spread -reuse-pid=13150 -discard                                                            
2018-07-27 17:43:01 Allocating google:amazon-linux-2-64...
2018-07-27 17:43:03 Cannot allocate google:amazon-linux-2-64: cannot allocate new Google server for amazon-linux-2-64: invalid value for field 'resource.disks[0].initializeParams.diskSizeGb': '10'. Requested disk size cannot be smaller than the image size (25 GB)                            
2018-07-27 17:43:03 Successful tasks: 0
2018-07-27 17:43:03 Aborted tasks: 1
error: unsuccessful run
```

Reply:
What should we do here? I would love to merge this PR - do we need to fix spread first before we can do that?

Reply:
spread is already fixed to support preserve-size at backend level. I am working to support this at system level but it needs gustavo's approval, so we can make this land and then make the change to add "storage: preserve-size" just for amazon-linux-2

Reply:
Sounds good to me!